app/controllers/application_controller.rb

```diff
@@ -2,4 +2,28 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  
+  def not_found
+    if env["REQUEST_PATH"] =~ /^\/api/
+      render :json => {:error => "not-found"}.to_json, :status => 404
```

Comment:
But talk of `rescue_from`, hope this wont rescue something else. Often am advised not to use it. Can you refactor this and well secured enough and let's see whats up?